pkg/volume/cinder/cinder.go

```diff
@@ -188,15 +192,21 @@ func (plugin *cinderPlugin) newProvisionerInternal(options volume.VolumeOptions,
 	}, nil
 }
 
-func getCloudProvider(cloudProvider cloudprovider.Interface) (CinderProvider, error) {
-	if cloud, ok := cloudProvider.(*openstack.OpenStack); ok && cloud != nil {
-		return cloud, nil
-	}
-	return nil, fmt.Errorf("wrong cloud type")
-}
-
 func (plugin *cinderPlugin) getCloudProvider() (CinderProvider, error) {
 	cloud := plugin.host.GetCloudProvider()
+	if cloud == nil {
+		if _, err := os.Stat(DefaultCloudConfigPath); err == nil {
+			var config *os.File
+			config, err = os.Open(DefaultCloudConfigPath)
+			if err != nil {
+				glog.V(4).Infof("unable to load openstack cloud provider : %v", err)
```

Comment:
Should this be an error log? Unless I'm mistaken, if there is an error opening the file and your log level is less than 4 then all you'll see is the `Cloud provider not initialized properly` error.

Reply:
good point fixing